lib/screenplay_web/auth_manager/error_handler.ex

```diff
@@ -9,8 +9,12 @@ defmodule ScreenplayWeb.AuthManager.ErrorHandler do
 
   @impl Guardian.Plug.ErrorHandler
   def auth_error(conn, error, _opts) do
-    auth_params = auth_params_for_error(error)
-    Phoenix.Controller.redirect(conn, to: ~p"/auth/keycloak?#{auth_params}")
+    if Plug.Conn.get_session(conn, :previous_path) =~ "api" do
```

Comment:
Shouldn't we be looking at the _current_ request path rather than the previous one? I'm surprised we set a "previous path" for API requests at all — I'd expect that value to always be some browser-navigable route.

Reply:
I think the main reason for holding onto the previous path is for the sake of auth redirects. When you log in to Keycloak, you're temporarily not at the URL you want to be at so we help the user get there by holding on to their original request path. I used it here just in case it landed in this function from a Keycloak request. Because `save_previous_path/2` is part of the request pipeline, it would be the same URL regardless of where the error surfaced.

Reply:
I think we should check the current request path in any case, given that sometimes we may not have a previous path or it may be a non-API page path (like for the first API request that happens after doing a full page load). "If current path is `api` _or_ previous path is `api`" in other words.